### PR TITLE
Adding index json for data generation used in fuse search

### PR DESCRIPTION
**Pull request by Era-cell**

### CODE REVIEW

**Comment by jviotti** [comment]
Don't you need the site config changes too for the JSON output to be generated?

File: layouts/_default/index.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{{- $.Scratch.Add "index" slice -}}
+{{- range .Site.RegularPages -}}
+    {{ $parts := split  .Permalink  "/" }}
+    {{ $dialect := index $parts 3 }}
+    {{ $vocabulary := index $parts 4 }}
+    {{- $.Scratch.Add "index" (dict "dialect" $dialect "vocabulary" $vocabulary "title" .Params.keyword "tags" .Params.kind "contents" .Plain "permalink" .Permalink) -}}
+{{- end -}}
+{{- $.Scratch.Get "index" | jsonify -}}
```
Comment (jviotti) [inline layouts/_default/index.json:8]: I think this file is missing a line ending at the end, so GitHub is showing this alert circle.

**Comment by Era-cell** [comment]
> Don't you need the site config changes too for the JSON output to be generated?

Correct. Let me add it

File: layouts/_default/index.json (hunk 1 above)
Comment (Era-cell) [reply layouts/_default/index.json:8]: I will need to read more about this. But in config.toml  file I shouldn't have an empty line at end right..
I think only in json file I should

Comment (jviotti) [reply layouts/_default/index.json]: Not an empty line but a line ending. That's sounds like an issue with your editor configuration. Not sure what editor you use, but try to make sure it has https://editorconfig.org interoperability! That should pick up the `.editorconfig` files I have in most of the repos and always do the right thing!

**Review by jviotti — APPROVED** [review]

**Comment by jviotti** [comment]
I fixed the line ending myself with a commit on top of your PR so we could merge. Other than that, looks and works great! Thanks!

File: layouts/_default/index.json (hunk 1 above)
Comment (Era-cell) [reply layouts/_default/index.json]: Then I think the formatter of my IDE did that. I get it a bit

**Pull request merged**